### ISSUE TITLE
Strange behaviour of the parameter validators

### ISSUE THREAD

**Issue by kpymkata** [cbeust/jcommander#101]
Hi,

We are having a strange behaviour from the way JCommander is dealing with validator classes. Let me describe our scenario in more details. We have a parent command, called Parent, and two completely disjoint subclasses - A and B, which are as well both JCommander commands.

Both A and B define parameter with the same name - call it 'par', of the same type (Parent does not have this 'par' parameter). But only command B specifies a validator class for this parameter.

Now, what happens is that we call the command A from the command line, and we get an error ... from the validator for command B! But this validator shouldn't be called at all for command A - this breaks our semantics and intention.

I made some debugging and this is what happens:

When JCommander launches, it scans and installs all recoginzed comands - for example {B, A, Parent}. Next, it enters some initializing phase of setting default values for all command parameters - it iterates the commands, one by one. First, it reaches the command B, and next it reaches its parameter 'par'. It tries to set default value to it, which is firstly checked to be taken from a property provider file (if any). At this point in time, such a value is being set to our parameter, because we do use a property provider file. Next, the validator is called for this default value. And here the validation fails - which is completely OK, since this value is not suitable for command B - but is perfectly OK for command A.

So, to summarize: although I call only one command, JCommander initializes and sets default values (and then calls validators) on all commands, and on all of their parameters.

To my understanding, this is a bug and it should be fixed.

Kindest Regards,
Krum.
